src/main/java/seedu/address/model/delivery/Delivery.java

```diff
@@ -6,60 +6,72 @@
 
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.model.person.Person;
+import seedu.address.model.product.Product;
 
 /**
  * Represents a Delivery in the application.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Delivery {
 
-    private final DeliveryDetail deliveryDetail;
-    // CHANGE TO SUPPLIER LATER ON
-    private final Person sender;
-
-    // Data fields
-    private final DeliveryStatus status;
-    private final DeliveryTime deliveryTime;
+    private final Product product;
+    private final Person sender; // CHANGE TO SUPPLIER LATER ON
+    private final Status status;
```

Comment:
I think this field cannot be set to final since on creation it will automatically be pending but it can be changed later on?